src/main/java/com/nike/cerberus/endpoints/secret/SecureDataEndpointV1.java

```diff
@@ -0,0 +1,204 @@
+
+    @Inject
+    protected SecureDataEndpointV1(SecureDataService secureDataService,
+                                   PermissionsService permissionService,
+                                   SafeDepositBoxService safeDepositBoxService) {
+
+        this.secureDataService = secureDataService;
+        this.permissionService = permissionService;
+        this.safeDepositBoxService = safeDepositBoxService;
+    }
+
+    public final CompletableFuture<ResponseInfo<O>> execute(RequestInfo<I> request,
+                                                            Executor longRunningTaskExecutor,
+                                                            ChannelHandlerContext ctx) {
+
+        final Optional<SecurityContext> securityContext =
+                CmsRequestSecurityValidator.getSecurityContextForRequest(request);
+
+        // If the token was invalid throw Vault style permission denied error
+        if (! securityContext.isPresent() || ! (securityContext.get().getUserPrincipal() instanceof CerberusPrincipal)) {
+            log.error("Security context was null or principal was not instance of Cerberus Principal");
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
+            );
+        }
+
+        SecureDataRequestInfo requestInfo = parseInfoFromPath(request.getPath());
+
+        // if that path was invalid throw Vault style permission denied error
+        if (! doesRequestHaveRequiredParams(requestInfo.getCategory(), requestInfo.getSdbSlug())) {
+            log.error("Required path params missing, PATH: {}", request.getPath());
+            return generateVaultStyleResponse(longRunningTaskExecutor,
+                    ctx,
+                    new VaultStyleErrorResponse.VaultStyleErrorResponseBuilder()
+                            .withError("permission denied")
+                            .build(),
+                    HttpResponseStatus.FORBIDDEN.code()
+            );
+        }
+
+        CerberusPrincipal principal = (CerberusPrincipal) securityContext.get().getUserPrincipal();
+
+        String sdbBasePath = String.format("%s/%s/", requestInfo.getCategory(), requestInfo.getSdbSlug());
+        Optional<String> sdbId = safeDepositBoxService.getSafeDepositBoxIdByPath(sdbBasePath);
+
+        // check that the principal has the proper permissions or throw Vault style permission denied error
+        if (! sdbId.isPresent() || ! permissionService.doesPrincipalHavePermission(principal, sdbId.get(),
+                        SecureDataAction.fromMethod(request.getMethod()))) {
+            log.info("SDB Id not found or permission was not granted for principal, principal: {}, sdb: {}," +
```

Comment:
I think we should log specifically "if it wasn't found" OR "if there wasn't permissions"

Reply:
- [ ] separate log statement